coordinator/src/position/models.rs

```diff
@@ -172,10 +173,10 @@ fn calculate_accept_settlement_amount(
         Direction::Short => short_leverage,
     };
 
-    let margin_trader_without_opening_fees = calculate_margin(opening_price, quantity, leverage);
+    let margin_trader = calculate_margin(opening_price, quantity, leverage);
 
-    let accept_settlement_amount = Decimal::from(margin_trader_without_opening_fees)
-        + Decimal::from(pnl)
+    let accept_settlement_amount = Decimal::from(margin_trader) + Decimal::from(pnl)
```

Comment:
I didn't notice this in the last PR but instead of doing normal math and then `max(Decimal::Zero)` we should use `checked_sub` here because otherwise this might overflow and we get a gigantic number.